src/NStore.Core/Processing/StreamProcessorExtensions.cs

```diff
@@ -1,13 +1,58 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
 using NStore.Core.Streams;
 
 namespace NStore.Core.Processing
 {
     public static class StreamProcessorExtensions
     {
-        public static StreamProcessor Fold(this IReadOnlyStream stream)
+        public static StreamProcessor Aggregate(this IReadOnlyStream stream)

```

Comment:
Maybe this could be called "CreateAggregator"? Actually all other AggregateXXX methods returns the result, this is the one that create the processor  and does not return a result, but an object you can interact with and finally call RunAsync to perform the aggregation.